metadatastore/core.py

```diff
@@ -1043,7 +1044,7 @@ def find_events(start_col, start_cache,
         # wrap it our fancy dict
         ev = doc.Document('Event', ev)
         yield ev
-
+    events.close()
```

Comment:
it is better to be explicit about closing down resources rather than relying on GC and presumably the `__del__` method to take care of it.